contracts/tokens/SecurityToken.sol

```diff
@@ -67,6 +68,40 @@ contract SecurityToken is ERC20, ERC20Detailed, Ownable, ReentrancyGuard, Securi
     //Event emit when the global treasury wallet address get changed
     event TreasuryWalletChanged(address _oldTreasuryWallet, address _newTreasuryWallet);
     event DisableController();
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+    event TokenUpgraded(uint8 _major, uint8 _minor, uint8 _patch);
+    /**
+     * @notice constructor
+     * @dev Never called as contract is created through Proxy factory
+     * @dev Needed otherwise contract is considered abstract
+     * @param _name Name of the SecurityToken
+     * @param _symbol Symbol of the Token
+     * @param _decimals Decimals for the securityToken
+     */
+    constructor(
```

Comment:
Why do we need to initialize this?

Reply:
ST inherits from ERC20Detailed, so we can't create the logic contract without calling the constructor of ERC20Detailed. We could refactor this if we wanted to, but probably not worth it IMO.